### PR TITLE
feat: adding helper functions, external support (mimir/cortex) + extras

### LINKED ISSUE #3: Expose PROMETHEUS_SERVER_ENDPOINT as a configurable value
There are scenarios where an operator may want to use a `prometheus` instance that does not run alongside `opencost`. We should provide the ability to define the `PROMETHEUS_SERVER_ENDPOINT` when deploying the helm chart.

### PR DESCRIPTION
**Pull request by ts-mini**
Hey!  There is a lot in here but I figured since I was making these changes to run opencost locally - might as well contribute upstream.  

I don't have a lot of opinions on yaml structure/arg structure - but just did what i thought would make sense.  

I'm currently using this deployed, and did a decent amount of testing - but lmk what you think/if this is an ok direction.

- closes #3

### CODE REVIEW

**Comment by Pokom** [comment]
Thank you @ts-mini, I love the contribution! I think this is the right direction and was able to render the charts locally, but haven't had a chance to run through deploying it to a local cluster.

Think the only thing I'd really ask is if you could either document how you generated the `values.yaml` documentation, or even better possibly add it as a CI step to detect changes and fail if the docs need to be updated.

File: README.md [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
+| opencost.ui.resources.limits.cpu | string | `"999m"` |  |
+| opencost.ui.resources.limits.memory | string | `"1G"` |  |
+| opencost.ui.resources.requests.cpu | string | `"10m"` |  |
+| opencost.ui.resources.requests.memory | string | `"55M"` |  |
+| service.annotations | object | `{}` |  |
+| service.labels | object | `{}` |  |
+| service.type | string | `"ClusterIP"` |  |
+| serviceAccount.annotations | object | `{}` |  |
+| serviceAccount.automountServiceAccountToken | bool | `true` |  |
+| serviceAccount.create | bool | `true` |  |
```
Comment (Pokom) [inline README.md:55]: This is awesome! Suggestion: Document how this was generated. I'm assuming something similar to (https://github.com/norwoodj/helm-docs) but just want to be certain.

**Review by Pokom — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ts-mini) [reply README.md:55]: it was in fact helm-docs, let me see if i can throw something together

**Comment by ts-mini** [comment]
@Pokom I've taken a stab at (i.e. copied a github action/linked) that does a pretty good job of productionalizing a chart like this w.r.t gh actions

I've also added a custom helm-docs template + a note on how to use it in the readme

File: README.md (hunk 1 above)
Comment (ts-mini) [reply README.md:65]: added https://github.com/opencost/opencost-helm-chart/pull/7/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R67

**Comment by mattray** [comment]
@ts-mini this is great, could you rebase to resolve the conflicts? We'll mark that it closes both https://github.com/opencost/opencost-helm-chart/issues/3 and https://github.com/opencost/opencost-helm-chart/issues/8

**Comment by ts-mini** [comment]
> @ts-mini this is great, could you rebase to resolve the conflicts? We'll mark that it closes both #3 and #8

rebased/squashed + tested and seems to be still good to go

**Comment by toscott** [comment]
@ts-mini Looks like there are still a couple linting errors in the values.yaml due to missing spaces.

**Comment by ts-mini** [comment]
> @ts-mini Looks like there are still a couple linting errors in the values.yaml due to missing spaces.
> 
> ----Edit--- Button click to fix :) [ts-mini#1](https://github.com/ts-mini/opencost-helm-chart/pull/1)

what a team effort

**Comment by mattray** [comment]
@ts-mini thanks!

**Pull request merged**